packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -1191,6 +1201,25 @@ export function deserializeResponseValue(
       } else {
         return `${restValue} as any`;
       }
+    case "enum":
+      if (!type.isFixed && !type.isNonExhaustive) {
+        return `${restValue} as ${type.name}`;
+      }
+      return restValue;
+    case "model":
+      if (type.discriminator) {
+        const discriminatorProp = type.properties?.filter(
+          (p) => p.restApiName === type.discriminator
+        );
+        if (
+          discriminatorProp?.length === 1 &&
+          discriminatorProp[0]?.type.isFixed === false &&
+          discriminatorProp[0].type.isNonExhaustive === false
+        ) {
+          return `${restValue} as ${type.name}`;
+        }
+      }
+      return restValue;
```

Comment:
@joheredi @dgetu this is a new logic related with the deserialization caused by we are not treating exhaustive enum the same in both RLC and Modular layer.

Reply:
Thanks! We'll use this as a guide when porting to the new serializers